stubs/Flask-SocketIO/flask_socketio/__init__.pyi

```diff
@@ -0,0 +1,97 @@
+from collections.abc import Callable
+from logging import Logger
+from threading import Thread
+from typing import Any
+from typing_extensions import NotRequired, ParamSpec, TypeAlias, TypedDict, Unpack
+
+from flask import Flask
+from flask.testing import FlaskClient
+from socketio.base_manager import BaseManager  # type: ignore[import]  # pyright: ignore[reportMissingImports]
+
+from .namespace import Namespace
+from .test_client import SocketIOTestClient
+
+_P = ParamSpec("_P")
+_ExceptionHandler: TypeAlias = Callable[[Exception], Any]
+_Handler: TypeAlias = Callable[[Any], Any]
+
+_HandlerDecorator: TypeAlias = Callable[[_Handler], _Handler]
+
+gevent_socketio_found: bool
+
+class _SocketIOConfig(TypedDict, total=False):
+    manage_session: NotRequired[bool]
+    message_queue: NotRequired[str]
+    channel: NotRequired[str]
+    path: NotRequired[str]
+    resource: NotRequired[str]
+
+    # SocketIO.Server options
+    client_manager: NotRequired[BaseManager]
+    logger: NotRequired[bool | Logger]
+    json: NotRequired[Any]
+    async_handlers: NotRequired[bool]
+    always_connect: NotRequired[bool]
```

Comment:
Using `total=False` means that all entries in the `TypeDict` are understood by the type checker as being not required. So if you're using `total=False`, there's no need to use `NotRequired[]` for every entry in the dictionary. You're just stating the same thing twice

Reply:
I didn't know that - thanks for clarifying! I must have got confused as I haven't used `NotRequired` before.